_posts/2020-3-27-SciML.md

```diff
@@ -0,0 +1,380 @@
+from mpl_toolkits.mplot3d import Axes3D
+fig = plt.figure()
+ax = fig.add_subplot(111, projection='3d')
+ax.plot(ut[0,:],ut[1,:],ut[2,:])
+plt.show()
+```
+
+### We provide tools for researching methods in scientific machine learning
+
+Last but not least, we support the research activities of practitioners in
+scientific machine learning. Tools like [DiffEqDevTools.jl](https://github.com/SciML/DiffEqDevTools.jl)
+and [RootedTrees.jl](https://github.com/SciML/RootedTrees.jl) make it easy to
+create and benchmark new methods and accelerate the publication
+process for numerical researchers. Our wrappers for external tools like
+[FEniCS.jl](https://github.com/SciML/FEniCS.jl) and
+[SciPyDiffEq.jl](https://github.com/SciML/SciPyDiffEq.jl) make it easy to perform
+cross-platform comparisons. Our stack is entirely written within Julia,
+which means every piece can be tweaked on the fly, making it easy to mix
+and match Hamiltonian integrators with neural networks to discover new scientific
+applications. Our issues and [chat channel](https://gitter.im/JuliaDiffEq/Lobby)
+serve as places to not just debug existing software, but also discuss new
+methods and help create high performance implementations.
+
+In addition, we support many student activities to bring new researchers into the
+community. Many of the maintainers of our packages, like Yingbo Ma, Vaibhav Dixit,
+Kanav Gupta, Kirill Zubov, etc. all started as one of our over 50 student developers
+from past [Google Summer of Code](https://summerofcode.withgoogle.com/) and other
+[Julia Seasons of Contributions](https://julialang.org/jsoc/).
+
+## The Goal of SciML
+
+When you read a paper that is
+[mixing neural networks with differential equations](https://arxiv.org/abs/2001.04385)
+or [designing new neural networks that satisfy incompressibility for modeling Navier-Stokes](https://arxiv.org/abs/1903.00033),
+you should be able to go online and find tweakable, high quality, and highly
+maintained package implementations of these methodologies to either start using
+for your scientific research, or utilize as a starting point for furthering the
+methods of scientific machine learning. For this reason, the goal of the SciML
+OSS organization is to be a hub for the development of robust cross-language
+scientific machine learning software. **In order to make this a reality, we as
+an organization have to commit to the following principles**:
+
+### Everything that we build is compatible with automatic differentiation
+
+Putting an arbitrary piece of code from the SciML group into a training loop of
+some machine learning library like [Flux](https://fluxml.ai/) will naturally work.
+This means we plan to enforce coding styles that are compatible with language-wide differentiable programming
+tools like [Zygote](https://github.com/FluxML/Zygote.jl), or provide pre-defined
+forward/adjoint rules via derivative rule package [ChainRules.jl](https://github.com/JuliaDiff/ChainRules.jl).

```

Comment:
```suggestion
forward/adjoint rules via the derivative rule package [ChainRules.jl](https://github.com/JuliaDiff/ChainRules.jl).

```
assuming you are specifically limiting this to ChainRules

Reply:
we aren't specifically limiting it to ChainRules, but we hope everything coalesces